keys/ranges.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package keys
+
+import "github.com/cockroachdb/cockroach/proto"
+
+// Range describes a kv range.
+type Range struct {
+	Start, End proto.Key
+}
+
+var (
+	// Meta1Range holds all first level addressing.
+	Meta1Range = Range{proto.KeyMin, Meta2Prefix}
+
+	// ZoneConfigRange is the range for all zone configs.
+	ZoneConfigRange = Range{ConfigZonePrefix, ConfigZonePrefix.PrefixEnd()}
+
+	// UserDataRange is the non-meta and non-structured portion of the key space.
+	UserDataRange = Range{SystemMax, TableDataPrefix}
+
+	// StructuredSystemRange is the range of system objects for structured data.
+	// It stops at the max reserved descriptor ID.
+	StructuredSystemRange = Range{TableDataPrefix, UserTableDataMin}
+
+	// NoSplitRanges describes the ranges that should never be split.
+	NoSplitRanges = []Range{Meta1Range, ZoneConfigRange, StructuredSystemRange}
```

Comment:
How come `StructuredSystemRange` can't be split? are you planning to lift this restriction when things are gossiped?